packages/platform-web/src/shuvi-app/react/AppContainer.tsx

```diff
@@ -22,12 +24,52 @@ function ErrorGuard({ children = null }: React.PropsWithChildren<{}>) {
   return <>{children}</>;
 }
 
+const uuid = () => {
+  return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function (c) {
+    var r = (Math.random() * 16) | 0,
+      v = c == 'x' ? r : (r & 0x3) | 0x8;
+    return v.toString(16);
+  });
+};
+
+function useNavigationTrace() {
+  const router = useRouter();
+  const route = useCurrentRoute();
+  let navigationTrace = React.useRef<typeof clientRenderTrace>();
+  React.useEffect(() => {
+    router.beforeEach((to, from, next) => {
+      navigationTrace.current = clientRenderTrace.traceChild(
+        'SHUVI_NAVIGATION_DONE'
+      );
+      const navigationTriggeredTrace = clientRenderTrace.traceChild(
+        'SHUVI_NAVIGATION_TRIGGERED'
+      );
+      const fromPath = `${from.pathname}${from.search}`;
+      const toPath = `${to.pathname}${to.search}`;
+      const navigationId = uuid();
+      navigationTrace.current.setAttribute('from', fromPath);
```

Comment:
I would like to know which trace has what attributes, maybe we should pre-define all of these in a place.